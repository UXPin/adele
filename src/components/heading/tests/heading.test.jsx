import { mount } from 'enzyme';
import React from 'react';
import 'jest-styled-components';

import Heading from '../heading';

it('renders Heading correctly when scroll is true', () => {
  const wrapper = mount(<Heading scroll={true} />);
  expect(wrapper).toMatchSnapshot();
});

it('renders Heading correctly when scroll is false', () => {
  const wrapper = mount(<Heading scroll={false} />);
  expect(wrapper).toMatchSnapshot();
});
