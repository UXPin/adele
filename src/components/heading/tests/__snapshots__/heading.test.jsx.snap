// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Heading correctly when scroll is false 1`] = `
.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  position: relative;
  height: 40px;
  min-height: 40px;
  width: 100%;
  max-width: 200px;
  background-color: #006cff;
  border-radius: 25px;
  border: 1px solid #006cff;
  font-family: proxima-nova,Proxima Nova,Helvetica Neue,Helvetica,Arial,sans-serif;
  font-size: 16px;
  font-weight: 300;
  color: white;
  -webkit-text-decoration: none;
  text-decoration: none;
  cursor: pointer;
  -webkit-transition: all ease-in 0.1s;
  transition: all ease-in 0.1s;
}

.c2:hover {
  background-color: #3389ff;
  border-color: #3389ff;
  box-shadow: 4px 4px 20px rgba(179,211,255,0.6);
}

.c2:active {
  background-color: #0061e6;
  box-shadow: 'none';
}

.c2:focus {
  outline: none;
  background-color: #3389ff;
}

.c3 {
  position: relative;
  outline: none;
  border: none;
  background: none;
  font-family: proxima-nova,Proxima Nova,Helvetica Neue,Helvetica,Arial,sans-serif;
  font-size: 16px;
  color: #777777;
  font-weight: 300;
  cursor: pointer;
}

.c3:hover {
  -webkit-text-decoration: underline;
  text-decoration: underline;
  -webkit-transition: all ease-in 0.2s;
  transition: all ease-in 0.2s;
}

.c3:focus {
  color: #006cff;
}

.c0 {
  font-family: proxima-nova,Proxima Nova,Helvetica Neue,Helvetica,Arial,sans-serif;
  margin: 150px auto 90px auto;
}

.c0 h1 {
  display: block;
  position: relative;
  margin: 0;
  font-size: 230px;
  text-align: center;
  -webkit-letter-spacing: -13px;
  -moz-letter-spacing: -13px;
  -ms-letter-spacing: -13px;
  letter-spacing: -13px;
  text-shadow: 10px 10px 20px rgba(0,0,55,0.07);
}

.c0 span {
  display: block;
  position: relative;
  width: 500px;
  margin: -10px auto 0 auto;
  font-size: 26px;
  font-weight: 300;
  line-height: 1.3;
  color: #777777;
  text-align: center;
  text-shadow: none;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  position: relative;
  margin-top: 40px;
}

.c1 > button:first-child {
  margin-right: 40px;
}

<Heading
  scroll={false}
>
  <styled.div
    scroll={false}
  >
    <div
      className="c0"
    >
      <h1>
        Adele
      </h1>
      <span>
        The repository of publicly available design systems and pattern libraries
      </span>
      <styled.section>
        <section
          className="c1"
        >
          <Button
            action={[Function]}
            label="Start Exploring Systems"
            tab={1}
            targetBlank={false}
            type="button"
          >
            <styled.button
              onClick={[Function]}
              tabIndex={1}
            >
              <button
                className="c2"
                onClick={[Function]}
                tabIndex={1}
              >
                Start Exploring Systems
              </button>
            </styled.button>
          </Button>
          <NavLink
            action={[Function]}
            label="Why Adele?"
            tab={1}
          >
            <styled.button
              onClick={[Function]}
              tabIndex={1}
            >
              <button
                className="c3"
                onClick={[Function]}
                tabIndex={1}
              >
                Why Adele?
              </button>
            </styled.button>
          </NavLink>
        </section>
      </styled.section>
    </div>
  </styled.div>
</Heading>
`;

exports[`renders Heading correctly when scroll is true 1`] = `
.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  position: relative;
  height: 40px;
  min-height: 40px;
  width: 100%;
  max-width: 200px;
  background-color: #006cff;
  border-radius: 25px;
  border: 1px solid #006cff;
  font-family: proxima-nova,Proxima Nova,Helvetica Neue,Helvetica,Arial,sans-serif;
  font-size: 16px;
  font-weight: 300;
  color: white;
  -webkit-text-decoration: none;
  text-decoration: none;
  cursor: pointer;
  -webkit-transition: all ease-in 0.1s;
  transition: all ease-in 0.1s;
}

.c2:hover {
  background-color: #3389ff;
  border-color: #3389ff;
  box-shadow: 4px 4px 20px rgba(179,211,255,0.6);
}

.c2:active {
  background-color: #0061e6;
  box-shadow: 'none';
}

.c2:focus {
  outline: none;
  background-color: #3389ff;
}

.c3 {
  position: relative;
  outline: none;
  border: none;
  background: none;
  font-family: proxima-nova,Proxima Nova,Helvetica Neue,Helvetica,Arial,sans-serif;
  font-size: 16px;
  color: #777777;
  font-weight: 300;
  cursor: pointer;
}

.c3:hover {
  -webkit-text-decoration: underline;
  text-decoration: underline;
  -webkit-transition: all ease-in 0.2s;
  transition: all ease-in 0.2s;
}

.c3:focus {
  color: #006cff;
}

.c0 {
  font-family: proxima-nova,Proxima Nova,Helvetica Neue,Helvetica,Arial,sans-serif;
  margin: 150px auto 90px auto;
}

.c0 h1 {
  display: block;
  position: relative;
  margin: 0;
  font-size: 230px;
  text-align: center;
  -webkit-letter-spacing: -13px;
  -moz-letter-spacing: -13px;
  -ms-letter-spacing: -13px;
  letter-spacing: -13px;
  text-shadow: 10px 10px 20px rgba(0,0,55,0.07);
}

.c0 span {
  display: block;
  position: relative;
  width: 500px;
  margin: -10px auto 0 auto;
  font-size: 26px;
  font-weight: 300;
  line-height: 1.3;
  color: #777777;
  text-align: center;
  text-shadow: none;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  position: relative;
  margin-top: 40px;
}

.c1 > button:first-child {
  margin-right: 40px;
}

<Heading
  scroll={true}
>
  <styled.div
    scroll={true}
  >
    <div
      className="c0"
    >
      <h1>
        Adele
      </h1>
      <span>
        The repository of publicly available design systems and pattern libraries
      </span>
      <styled.section>
        <section
          className="c1"
        >
          <Button
            action={[Function]}
            label="Start Exploring Systems"
            tab={1}
            targetBlank={false}
            type="button"
          >
            <styled.button
              onClick={[Function]}
              tabIndex={1}
            >
              <button
                className="c2"
                onClick={[Function]}
                tabIndex={1}
              >
                Start Exploring Systems
              </button>
            </styled.button>
          </Button>
          <NavLink
            action={[Function]}
            label="Why Adele?"
            tab={1}
          >
            <styled.button
              onClick={[Function]}
              tabIndex={1}
            >
              <button
                className="c3"
                onClick={[Function]}
                tabIndex={1}
              >
                Why Adele?
              </button>
            </styled.button>
          </NavLink>
        </section>
      </styled.section>
    </div>
  </styled.div>
</Heading>
`;
